Zend/zend_compile.c

```diff
@@ -3020,9 +3145,11 @@ void zend_compile_assign(znode *result, zend_ast *ast) /* {{{ */
 			}
 
 			zend_compile_list_assign(result, var_ast, &expr_node, var_ast->attr);
-			return;
+			break;
 		EMPTY_SWITCH_DEFAULT_CASE();
 	}
+
+	CG(short_circuiting) = 1;
```

Comment:
You have some code that restores short_circuiting to what it was before, and some that just sets it to 1 afterwards. I don't really understand under what circumstances which is appropriate.

Reply:
That is confusing. In some cases it is given that you're in a short-circuiting chain but for consistency it might make more sense to add the check anyway.

Reply:
The code is now consistently the same.